### ISSUE TITLE
Can't make a hist out of a Geant4 histogram

### ISSUE THREAD

**Issue by henryiii** [scikit-hep/uproot5#883]
When trying to make a Hist out of an geant4 histogram, you get the following error:

```python
import uproot
uproot_file = uproot.open("../scikit-hep-testdata/src/skhep_testdata/data/uproot-from-geant4.root")
uproot_file["cot_diff"].to_hist()
```

```pytb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/henryschreiner/git/scikit-hep/uproot-browser/.venv/lib/python3.11/site-packages/uproot/behaviors/TH1.py", line 277, in to_hist
    self.to_boost(metadata=metadata, axis_metadata=axis_metadata)
  File "/Users/henryschreiner/git/scikit-hep/uproot-browser/.venv/lib/python3.11/site-packages/uproot/behaviors/TH1.py", line 226, in to_boost
    axes = [
           ^
  File "/Users/henryschreiner/git/scikit-hep/uproot-browser/.venv/lib/python3.11/site-packages/uproot/behaviors/TH1.py", line 227, in <listcomp>
    _boost_axis(self.member(name), axis_metadata)
  File "/Users/henryschreiner/git/scikit-hep/uproot-browser/.venv/lib/python3.11/site-packages/uproot/behaviors/TH1.py", line 23, in _boost_axis
    if axis.member("fLabels") is not None:
       ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/henryschreiner/git/scikit-hep/uproot-browser/.venv/lib/python3.11/site-packages/uproot/model.py", line 556, in member
    raise uproot.KeyInFileError(
uproot.exceptions.KeyInFileError: not found: 'fLabels' because uproot.dynamic.Model_TAxis_v6 has only the following members:

    '@fUniqueID', '@fBits', 'fName', 'fTitle', 'fNdivisions', 'fAxisColor', 'fLabelColor', 'fLabelFont', 'fLabelOffset', 'fLabelSize', 'fTickLength', 'fTitleOffset', 'fTitleSize', 'fTitleColor', 'fTitleFont', 'fNbins', 'fXmin', 'fXmax', 'fXbins', 'fFirst', 'fLast', 'fTimeDisplay', 'fTimeFormat'

in file ../scikit-hep-testdata/src/skhep_testdata/data/uproot-from-geant4.root
```

You can see this in the test data files, uproot-from-geant4.root. `cot_diff` is an example histogram that fails.

Using Uproot 5.0.7.

I think the probe is `axis.member("fLabels")` throws an error, not returns `None`, when it's not present, but this code assumes it returns None. This should probably have `none_if_missing=True`. (Just curious, why is this not `default=None`? That seems more pythonic, would look like `dict.get`).

**Comment by jpivarski** [comment]
> I think the probe is `axis.member("fLabels")` throws an error, not returns `None`, when it's not present, but this code assumes it returns None. This should probably have `none_if_missing=True`.

Your assessment is entirely correct. The `none_if_missing` argument was added to make it easier to write code for ROOT classes of different versions, which may be missing some attributes. (Geant4 has its own ROOT writer, so its class versions may be very old.) This is such a case.

> (Just curious, why is this not `default=None`? That seems more pythonic, would look like `dict.get`).

That would have been a much better name. This was developed as part of a coding sprint, not carefully crafting a public API. I can see reasons why downstream developers would need to use this function (to write frameworks for their custom C++ classes), so deprecating `none_if_missing` (boolean) in favor of `default` (any value) may be a good idea. First, I'd like to scan all those GitHub repos I have on my computer at home to see how often it is actually used. We don't have a deprecation policy in place for Uproot as we do for Awkward, since the API is not rapidly changing. (Most changes are bug-fixes that keep the intended API unchanged.)

**Comment by jpivarski** [comment]
Just a note: we'll need to check this again—maybe the Uproot issue is solved by https://github.com/scikit-hep/uproot5/pull/884.

**Comment by ioanaif** [comment]
Tested with 

```
>>> import uproot
>>> import skhep_testdata
>>> uproot_file = uproot.open(skhep_testdata.data_path("uproot-from-geant4.root"))
>>> uproot_file["cot_diff"].to_hist()
Hist(Regular(200, -0.266667, 0.266667, name='xaxis', label='#delta cot #theta'), storage=Weight()) # Sum: WeightedSum(value=220, variance=220) (WeightedSum(value=224, variance=224) with flow)
```

Seems to be solved by #884
